api/src/main/java/io/opentelemetry/common/AttributesKeys.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.common;
+
+import java.util.List;
+
+/**
+ * Creation methods for {@link AttributeKey} implementations.
+ *
+ * @see Attributes
+ */
+public class AttributesKeys {
+  private AttributesKeys() {}
+
+  /** Create a new AttributeKey for String valued attributes. */
+  public static AttributeKey<String> stringKey(String key) {
+    return AttributeKeyImpl.create(key, AttributeType.STRING);
+  }
+
+  /** Create a new AttributeKey for Boolean valued attributes. */
+  public static AttributeKey<Boolean> booleanKey(String key) {
+    return AttributeKeyImpl.create(key, AttributeType.BOOLEAN);
+  }
+
+  /** Create a new AttributeKey for Long valued attributes. */
+  public static AttributeKey<Long> longKey(String key) {
+    return AttributeKeyImpl.create(key, AttributeType.LONG);
+  }
+
+  /** Create a new AttributeKey for Double valued attributes. */
+  public static AttributeKey<Double> doubleKey(String key) {
+    return AttributeKeyImpl.create(key, AttributeType.DOUBLE);
+  }
+
+  /** Create a new AttributeKey for List&lt;String&gt; valued attributes. */
+  public static AttributeKey<List<String>> stringArrayKey(String key) {
+    return AttributeKeyImpl.create(key, AttributeType.STRING_ARRAY);
```

Comment:
Just had an idea which I think could have been possible with separate key impls but don't think so with AutoValue version. If the array keys implemented a marker interface like `AttributeArrayKey<T>` then we could also provide `setAttribute(AttributeArrayKey<T>, T...)`. Array keys aren't so common though and we could rework later too so no big deal now though.

Reply:
yeah, I don't think we have any semantic conventions for array attributes, so I'm much less worried about them. Let's punt on this for now, and revisit if array attributes really become important.